### PR TITLE
Update ANCM doc for 3.0

### LINKED ISSUE #12843: Update ANCM doc for 3.0 (Generic Host)
Web Host --&gt; Generic Host

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: fe69b3a9-896f-e189-0c2c-f87ff735df9c
* Version Independent ID: 97c8549b-66ad-0a23-1971-784fb923042a
* Content: [ASP.NET Core Module](https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-2.2#additional-resources)
* Content Source: [aspnetcore/host-and-deploy/aspnet-core-module.md](https://github.com/aspnet/AspNetCore.Docs/blob/master/aspnetcore/host-and-deploy/aspnet-core-module.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-hostdeploy**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by guardrex**
Fixes  #12843

~[Internal Review Topic 2.1](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-2.1&branch=pr-en-us-14462)~ **no or few changes**
[Internal 3.0 version](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-3.0&branch=pr-en-us-14462)

* Whole-topic versioning.
* For 3.0 updates:
  * A couple of link changes and an API reference update.
  * Changes for the default hosting model.

@halter73, should `shirhatti` look?

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@halter73  can you take a quick review of [Internal 3.0 version](https://review.docs.microsoft.com/en-us/aspnet/core/host-and-deploy/aspnet-core-module?view=aspnetcore-3.0&branch=pr-en-us-14462) 
That's where all the changes are.

**Comment by Rick-Anderson** [comment]
@halter73 can you take a quick review of Internal 3.0 version or recommend another reviewer?
That's where all the changes are.

File: aspnetcore/host-and-deploy/aspnet-core-module.md [hunk 1]
```diff
@@ -660,6 +415,617 @@ The Hosting Bundle installer logs for the module are found at *C:\\Users\\%UserN
...
+* Forward web requests to a backend ASP.NET Core app running the [Kestrel server](xref:fundamentals/servers/kestrel), called the [out-of-process hosting model](#out-of-process-hosting-model).
+
+Supported Windows versions:
+
+* Windows 7 or later
+* Windows Server 2008 R2 or later
+
+When hosting in-process, the module uses an in-process server implementation for IIS, called IIS HTTP Server (`IISHttpServer`).
+
+When hosting out-of-process, the module only works with Kestrel. The module is incompatible with [HTTP.sys](xref:fundamentals/servers/httpsys).
```
Comment (halter73) [inline aspnetcore/host-and-deploy/aspnet-core-module.md]: Maybe this should explain what it means to be incompatible with HTTP.sys, because I'm not really sure what we're trying to indicate when we say here.

File: aspnetcore/host-and-deploy/aspnet-core-module.md [hunk 2]
```diff
@@ -660,6 +415,617 @@ The Hosting Bundle installer logs for the module are found at *C:\\Users\\%UserN
...
+
+* The [requestTimeout attribute](#attributes-of-the-aspnetcore-element) doesn't apply to in-process hosting.
+
+* Sharing an app pool among apps isn't supported. Use one app pool per app.
+
+* When using [Web Deploy](/iis/publish/using-web-deploy/introduction-to-web-deploy) or manually placing an [app_offline.htm file in the deployment](xref:host-and-deploy/iis/index#locked-deployment-files), the app might not shut down immediately if there's an open connection. For example, a websocket connection may delay app shut down.
+
+* The architecture (bitness) of the app and installed runtime (x64 or x86) must match the architecture of the app pool.
+
+* If setting up the app's host manually with `WebHostBuilder` (not using [CreateDefaultBuilder](xref:fundamentals/host/web-host#set-up-a-host)) and the app is ever run directly on the Kestrel server (self-hosted), call `UseKestrel` before calling `UseIISIntegration`. If the order is reversed, the host fails to start.
```
Comment (halter73) [inline aspnetcore/host-and-deploy/aspnet-core-module.md]: I'm pretty sure the comment about how reversing the order of `UseKestrel` and `UseIISIntegration` breaks things hasn't been true for a while. You fixed that, right @Tratcher?

Comment (halter73) [reply aspnetcore/host-and-deploy/aspnet-core-module.md]: Also, does this really belong in the "In-process hosting model" section? I guess this is about self-hosting apps that are normally hosted in-proc by ancm, so it makes a little bit a sense, but it's also a little confusing.

Comment (Tratcher) [reply aspnetcore/host-and-deploy/aspnet-core-module.md]: Yeah, we dealt with that a long time ago. Maybe it was this one? https://github.com/aspnet/IISIntegration/issues/242

File: aspnetcore/host-and-deploy/aspnet-core-module.md [hunk 3]
```diff
@@ -513,12 +309,8 @@ The handler settings can also be provided via environment variables:
 > [!WARNING]
 > Do **not** leave debug logging enabled in the deployment for longer than required to troubleshoot an issue. The size of the log isn't limited. Leaving the debug log enabled can exhaust the available disk space and crash the server or app service.
 
-::: moniker-end
-
 See [Configuration with web.config](#configuration-with-webconfig) for an example of the `aspNetCore` element in the *web.config* file.
 
-::: moniker range=">= aspnetcore-3.0"
-
 ## Modify the stack size
```
Comment (halter73) [inline aspnetcore/host-and-deploy/aspnet-core-module.md]: It might be worth pointing out this only applies to the InProcess hosting model.

File: aspnetcore/host-and-deploy/aspnet-core-module.md [hunk 4]
```diff
@@ -660,6 +415,617 @@ The Hosting Bundle installer logs for the module are found at *C:\\Users\\%UserN
...
+
+  For sample code that sets the app's current directory, see the [CurrentDirectoryHelpers class](https://github.com/aspnet/AspNetCore.Docs/tree/master/aspnetcore/host-and-deploy/aspnet-core-module/samples_snapshot/2.x/CurrentDirectoryHelpers.cs). Call the `SetCurrentDirectory` method. Subsequent calls to <xref:System.IO.Directory.GetCurrentDirectory*> provide the app's directory.
+
+* When hosting in-process, <xref:Microsoft.AspNetCore.Authentication.AuthenticationService.AuthenticateAsync*> isn't called internally to initialize a user. Therefore, an <xref:Microsoft.AspNetCore.Authentication.IClaimsTransformation> implementation used to transform claims after every authentication isn't activated by default. When transforming claims with an <xref:Microsoft.AspNetCore.Authentication.IClaimsTransformation> implementation, call <xref:Microsoft.Extensions.DependencyInjection.AuthenticationServiceCollectionExtensions.AddAuthentication*> to add authentication services:
```
Comment (halter73) [inline aspnetcore/host-and-deploy/aspnet-core-module.md]: This now reads exactly the same as the "When hosting out-of-process" section about AuthenticateAsync.

File: aspnetcore/host-and-deploy/aspnet-core-module.md [hunk 5]
```diff
@@ -660,6 +415,617 @@ The Hosting Bundle installer logs for the module are found at *C:\\Users\\%UserN
...
+> [!WARNING]
+> Do **not** leave debug logging enabled in the deployment for longer than required to troubleshoot an issue. The size of the log isn't limited. Leaving the debug log enabled can exhaust the available disk space and crash the server or app service.
+
+See [Configuration with web.config](#configuration-with-webconfig) for an example of the `aspNetCore` element in the *web.config* file.
+
+## Proxy configuration uses HTTP protocol and a pairing token
+
+*Only applies to out-of-process hosting.*
+
+The proxy created between the ASP.NET Core Module and Kestrel uses the HTTP protocol. Using HTTP is a performance optimization, where the traffic between the module and Kestrel takes place on a loopback address off of the network interface. There's no risk of eavesdropping the traffic between the module and Kestrel from a location off of the server.
```
Comment (halter73) [inline aspnetcore/host-and-deploy/aspnet-core-module.md]: > Using HTTP is a performance optimization,

I don't think using HTTP is really a performance optimization. It's just what we happen to use. I would rephrase:

```suggestion
The proxy created between the ASP.NET Core Module and Kestrel uses the HTTP protocol where the traffic between the module and Kestrel takes place on a loopback address. There's no risk of eavesdropping the traffic between the module and Kestrel from a location off of the server.
```

Comment (Tratcher) [reply aspnetcore/host-and-deploy/aspnet-core-module.md]: It's a design choice. We can exclude that line. The security risk line is the important part, people have asked about that.
